feat: add tool selection by feature param

MQ37 opened the pull request:
closes https://github.com/apify/actors-mcp-server/issues/166

**Only `docs` tools category is currently enabled by default (always)**

This PR introduces the `tools` (`--tools` or `?tools`) comma separated list param so users can choose which tools are loaded into their MCP session.

Possible values:
- docs: Search and fetch Apify documentation tools.
- runs: Get Actor runs list, run details, and logs from a specific Actor run.
- storage: Access datasets, key-value stores, and their records.

Example:
```
--tools docs,runs
?tools=docs,storage
```

Also vibe refactored the tool loading logic so it's not duplicated in the `stdio.ts`

jirispilka commented on src/tools/index.ts:
```diff
@@ -1,45 +1,53 @@
-    fetchApifyDocsTool,
 ];
 
+export const featureTools = {
```
I don't have strong opinion but what about toolCategories? I think it is more descriptive?

jirispilka commented on src/tools/index.ts:
```diff
@@ -1,45 +1,53 @@
+    ],
+};
+
 export const betaTools = [
```
should we include betaTools into the toolsCategories? I think it would simplify the logic, why to have to switches for the same thing? If we want we can call it preview?

jirispilka reviewed (approved):
Looks good to me. The only change I would do is rename things and remove the beta flag. What do you think?

MQ37 commented:
> Looks good to me. The only change I would do is rename things and remove the beta flag. What do you think?

Agree, renamed and moved the beta tools to `preview` tools category :+1: Thank you for review

MichalKalita commented on src/stdio.ts:
```diff
@@ -88,8 +102,18 @@ if (!process.env.APIFY_TOKEN) {
+    };
+
+    // Use the shared tools loading logic
+    const tools = await loadToolsFromInput(input, process.env.APIFY_TOKEN as string, actorList.length === 0);
```
Missing validation if tools list is only list of available options.

MQ37 replied:
When the user inputs invalid option it just ignores it. What is goal of validation here? Do you think we should refuse to start the MCP server and exit with error code when the option is not valid?

MichalKalita commented on src/mcp/server.ts:
```diff
@@ -172,7 +166,14 @@ export class ActorsMcpServer {
-        const internalToolMap = new Map([...defaultTools, ...addRemoveTools, ...betaTools].map((tool) => [tool.tool.name, tool]));
+        const internalToolMap = new Map([...defaultTools, ...addRemoveTools].map((tool) => [tool.tool.name, tool]));
+        // Add all category tools
+        for (const key of Object.keys(toolCategories)) {
```
`Object.values(toolCategories).flat()` will be better here. Key is used only for get value from object.
